syntax = "proto3";
package user;

service UserService {
  rpc GetProfile (GetProfileRequest) returns (ProfileReply);
  rpc GetProfileByUserId (GetByUserIdRequest) returns (ProfileReply);
}

message GetProfileRequest {
  string token = 1;
}

message GetByUserIdRequest {
  int32 user_id = 1;
}

message ProfileReply {
  int32 id = 1;
  int32 user_id = 2;
  string username = 3;
  string display_name = 4;
  string bio = 5;
  string avatar_url = 6;
  bool is_private = 7;
}